lib/config.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+
+let _       = require('lodash'),
+    path    = require('path');
+
+class Config {
+
+    constructor(pkg, options, basePath) {
+        let legacyLayouts;
+        this.options = options;
+        this.basePath = basePath;
+        this.cacheControlHeader = this.options.maxAge || process.env.CACHE_CONTROL || 'max-age=60';
+        this.customHeaders = this.options.customHeaders || [];
+        this.description = this.options.description || pkg.description;
+        this.env = this.options.environment || process.env.ENVIRONMENT || process.env.NODE_ENV || 'development';
+        this.envProd = this.options.envProd || 'production';
+        this.port = this.options.port || process.env.PORT || 3000;
+        this.host = this.options.host || process.env.HOST || '0.0.0.0';
+        this.name = this.options.name || pkg.name;
+        this.surrogateControlHeader = this.options.surrogateCacheControl || process.env.SURROGATE_CACHE_CONTROL || 'max-age=360, stale-while-revalidate=60, stale-if-error=86400';
+        this.contactEmail = this.options.contactEmail || undefined;
+        this.contactName = this.options.contactName || undefined;
+        this.contactUrl = this.options.contactUrl || undefined;
+        this.licenseName = this.options.licenseName || undefined;
+        this.licenseUrl = this.options.licenseUrl || undefined;
+        this.termsOfService = this.options.termsOfService || undefined;
+        this.maxListeners = this.options.maxListeners || 10;
```

Comment:
@codeclout Looked through - must have remembered wrong.  Rest of the way it's done looks good.